pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -271,19 +276,63 @@ func (m *ManagerImpl) isVersionCompatibleWithPlugin(versions []string) bool {
 func (m *ManagerImpl) Allocate(node *schedulercache.NodeInfo, attrs *lifecycle.PodAdmitAttributes) error {
 	pod := attrs.Pod
 	devicesToReuse := make(map[string]sets.String)
+	// key by resource name, representing all the devices to be allocated of one resource
+	resourceDevices := make(map[string]sets.String)
+
+	// step1. aggregate all devices requested by containers
 	for _, container := range pod.Spec.InitContainers {
-		if err := m.allocateContainerResources(pod, &container, devicesToReuse); err != nil {
+		if devices, err := m.allocateContainerResources(pod, &container, devicesToReuse); err != nil {
 			return err
+		} else {
```

Comment:
please replace:
```
} else {
	// union devices to the underlying resource
	unionDevices(resourceDevices, devices)
}
```

with:
```
}
// union devices to the underlying resource
unionDevices(resourceDevices, devices)
```

Reply:
I am still not sure how feasible this modification could be. I will make more optimizations after other reviews :)